jerry-core/vm/vm.c

```diff
@@ -3416,28 +3416,35 @@ vm_execute (vm_frame_ctx_t *frame_ctx_p, /**< frame context */
   {
     completion_value = vm_loop (frame_ctx_p);
 
-    if (frame_ctx_p->call_operation == VM_NO_EXEC_OP)
-    {
-      break;
-    }
-
-    if (frame_ctx_p->call_operation == VM_EXEC_CALL)
+    switch (frame_ctx_p->call_operation)
     {
-      opfunc_call (frame_ctx_p);
-    }
+      case VM_NO_EXEC_OP:
+      {
+        goto exec_end;
```

Comment:
What about moving the code after exec_end here? We would not need any goto then.